packages/amplify-e2e-core/src/configure/index.ts

```diff
@@ -41,42 +50,48 @@ const authenticationOptions = ['AWS profile', 'AWS access keys'];
 
 const MANDATORY_PARAMS = ['accessKeyId', 'secretAccessKey', 'region'];
 
-export function amplifyConfigure(settings: AmplifyConfiguration): Promise<void> {
-  const s = { ...defaultSettings, ...settings };
-  const missingParam = MANDATORY_PARAMS.filter((p) => !Object.keys(s).includes(p));
-  if (missingParam.length) {
-    throw new Error(`mandatory params ${missingParam.join(' ')} are missing`);
-  }
-
-  return new Promise((resolve, reject) => {
-    const chain = spawn(getCLIPath(), ['configure'], { stripColors: true })
-      .wait('Sign in to your AWS administrator account:')
-      .wait('Press Enter to continue')
-      .sendCarriageReturn()
-      .wait('Specify the AWS Region');
-
-    singleSelect(chain, s.region, amplifyRegions);
+export function amplifyConfigure(cwd: string | null, settings: AmplifyConfiguration | null, flag?: CommandFlags): Promise<void> {
```

Comment:
If `settings` and `flag` are mutually exclusive (they are by looking at implementation) then please either change api to have union type (better) OR add validation to this function and throw if both are provided (not bad, but longer feedback loop than compiler error).

Yet other option is to split this function into two `amplifyConfigure` as is originally and `amplifyConfigureTelemetryFlags`.


```suggestion
export function amplifyConfigure(cwd: string | null, settings: AmplifyConfiguration | CommandFlags | null): Promise<void> {
```

Reply:
They are not really "mutually exclusive". Maybe at this moment, but there are several scenarios we may need both
- if we add tests for subcommands without creating a new function (e.g. `amplifyConfigureProject`)
- if we add a new flag with requires an option

So I think It's more scalable to separate them into 2 different variables.

Ideally, it'd be even better if we could use a destracture IMO

Reply:
`settings` are not consumed if `flag` is passed at all. They are mutually exclusive effectively.
Besides scenarios in both logical branches in that if conditions are never going to be combined together. I.e. you won't ever set telemetry flags AND configure profiles.

```
if (flag) {
    return spawn(getCLIPath(), ['configure', `--${flag}`], { cwd, stripColors: true })
      .wait(commandFlagsReturnMessage[flag])
      .runAsync();
  } else {
    const allSettings = { ...defaultSettings, ...settings };
    const missingParam = MANDATORY_PARAMS.filter((p) => !Object.keys(allSettings).includes(p));
```

Reply:
> So I think It's more scalable to separate them into 2 different variables.

Define `scalable`

Reply:
This is classic SRP problem.

Reply:
I see. That's fair. Done 049cdcd7cec03d1d33f979861284b2884258417b